### DISCUSSION TITLE
Migration path from v1.1.1 to 3.x?

### DISCUSSION THREAD

**Discussion by aldehoff** [Kozea/Radicale#1361] [Q&A general]
I've been using Radicale as my personal CalDAV and CardDAV server for many 
years. I like it a lot because it is simple and does its job. I haven't had 
much hassle with it apart from trying to upgrade from my old installation of 
version 1.1.1 to 2.1 a long time ago. I gave up on it after a few days because 
I stumbled upon two bugs

https://github.com/Kozea/Radicale/issues/830
https://github.com/Kozea/Radicale/issues/1007

that messed with my existing database. I just couldn't get it to work, so I 
pinned version 1.1.1 in my Ansible playbook and never looked back. The 
installation has continued to work just fine, and my address book and calendar 
have grown over time.

Now there's a new Debian Stable out, and it has Radicale 3.x. I'm tempted to 
give it another try, but the documentation states that I will still have to do 
that troublesome migration from 1.x to 2.x. Those two bugs are still open, and 
I don't know what to do.

Is there a cleaner upgrade path now that fixes the problems shown in those two 
bugs? Or am I doomed to keep my trusty old v1.1.1 running until the end of 
days?

Thanks for your time and your work on Radicale!

**Comment by jonassmedegaard** [comment]
Radicale used to be more tolerant. Newer releases are more strict. Both those referenced bugreports reflect that.

Radicale never was a tool for cleaning dirty data, and I would not put my hopes up on Radicale growing such feature, but also would not give up. Instead I would roll up my sleeves and cleanup my data independently from Radicale.

**Comment by jonassmedegaard** [comment]
In the past when needing to clean messy iCalendar data, I have had good use of the command-line tool icalmerge from the project [ical2html](https://www.w3.org/Tools/Ical2html/) - if you use Debian then I've packaged it there as package `ical2html`.
